roles/openshift_logging/tasks/set_defaults_from_current.yml

```diff
@@ -11,24 +11,236 @@
+        __openshift_logging_fluentd_buffer_size_limit: "{{ openshift_logging_facts['fluentd']['daemonsets']['logging-fluentd']['containers'][0]['env'] | entry_from_name_value_pair('BUFFER_SIZE_LIMIT') }}"
+
+    - conditional_set_fact:
+        facts: "{{ openshift_logging_facts['fluentd']['daemonsets']['logging-fluentd']['containers'][0] | flatten_dict }}"
+        vars:
+          __openshift_logging_fluentd_cpu_limit: resources.limits.cpu
+          __openshift_logging_fluentd_memory_limit: resources.limits.memory
+          __openshift_logging_fluentd_cpu_request: resources.requests.cpu
+
+
+# Mux
+- when: openshift_logging_facts['mux']['deploymentconfigs']['logging-mux'] is defined
+  block:
+    - set_fact:
+        __openshift_logging_mux_file_buffer_limit: "{{ openshift_logging_facts['mux']['deploymentconfigs']['logging-mux']['containers']['mux']['env'] | entry_from_name_value_pair('FILE_BUFFER_LIMIT') }}"
+        __openshift_logging_mux_buffer_queue_limit: "{{ openshift_logging_facts['mux']['deploymentconfigs']['logging-mux']['containers']['mux']['env'] | entry_from_name_value_pair('BUFFER_QUEUE_LIMIT') }}"
+        __openshift_logging_mux_buffer_size_limit: "{{ openshift_logging_facts['mux']['deploymentconfigs']['logging-mux']['containers']['mux']['env'] | entry_from_name_value_pair('BUFFER_SIZE_LIMIT') }}"
+
+    - conditional_set_fact:
+        facts: "{{ openshift_logging_facts['mux']['deploymentconfigs']['logging-mux']['containers'] | flatten_dict }}"
+        vars:
+          __openshift_logging_mux_cpu_limit: mux.resources.limits.cpu
+          __openshift_logging_mux_memory_limit: mux.resources.limits.memory
+          __openshift_logging_mux_cpu_request: mux.resources.requests.cpu
+
+
+# EventRouter
+- when: openshift_logging_facts['eventrouter']['deploymentconfigs']['logging-eventrouter'] is defined
+  conditional_set_fact:
+    facts: "{{ openshift_logging_facts['eventrouter']['deploymentconfigs']['logging-eventrouter']['containers'] | flatten_dict }}"
+    vars:
+      __openshift_logging_eventrouter_cpu_limit: kube-eventrouter.resources.limits.cpu
+      __openshift_logging_eventrouter_cpu_request: kube-eventrouter.resources.requests.cpu
+      __openshift_logging_eventrouter_memory_limit: kube-eventrouter.resources.limits.memory
+
+
+# Set the defaults based on collected facts
 - conditional_set_fact:
     facts: "{{ hostvars[inventory_hostname] }}"
     vars:
+      # Elasticsearch
       openshift_logging_es_number_of_shards: openshift_logging_es_number_of_shards | __openshift_logging_es_number_of_shards
       openshift_logging_es_number_of_replicas: openshift_logging_es_number_of_replicas | __openshift_logging_es_number_of_replicas
+      openshift_logging_elasticsearch_recover_after_time: openshift_logging_es_recover_after_time | __openshift_logging_es_recover_after_time
```

Comment:
I see we use this same format from before but doesnt this say 'default over fact'? Isnt this a 'short-circuit' or statement?

Reply:
It should only be looking at the inventory variables in this case because the fact source for this is `{{hostvars[inventory_hostname]}}`

@sdodson can you sanity check me on this?

Reply:
@jcantrill I confirmed this with a debug statement. Those variables aren't available in `"{{ hostvars[inventory_hostname] }}"` unless they have been set in the inventory or by `set_fact`